### PR TITLE
Create a Joomla 5 build

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This aims to add a joomla 5 dismaker dist

Before
----------------------------------------
No zip specific to Joomla 5

After
----------------------------------------
Zip that can support Joomla 5

@totten

### CODE REVIEW

**Comment by aydun** [comment]
Thanks for working on this @seamuslee001   Just checking that you have seen the later comments on https://lab.civicrm.org/dev/joomla/-/issues/54  - to create a separate J5 build and not to do the double compression (zip within a zip).

**Comment by seamuslee001** [comment]
@aydun this shoul be following the "Alt" build style and not doing the double zip

File: distmaker/distmaker.sh [hunk 1]
```diff
@@ -178,11 +179,15 @@ case $1 in
...
-  j5|Joomla)
+  Joomla)
   echo; echo "Generating Joomla PHP5 module"; echo;
   J5PACK=1
   ;;
 
+  j5|Joomla5)
+  echo; echo "Generating Joomla 5 PHP Module"; echo;
+  JOOMLA5PACK=1
+
```
Comment (totten) [inline distmaker/distmaker.sh:190]: @seamuslee001 I've just implemented a label `run-distmaker` (*which generates tarballs from a PR*) and tried here a test-case. If you check results on `CiviCRM-Distmaker-PR`, you should be able to see this console error:

```
  j5|Joomla5)
  echo; echo "Generating Joomla 5 PHP Module"; echo;
  JOOMLA5PACK=1

  # WORDPRESS PHP5
  wp5|WordPress)
./distmaker.sh: line 192: syntax error near unexpected token `)'
[chrome-headless] Kill process (pid=4474, sig=15)
[mysql] Kill process (pid=4258, sig=15)
```

It's missing `;;` at the end of Joomla5 block.

(Aside: Hmm, after fixing that, there's another issue... tester is [not going to run the new `Joomla5` target](https://github.com/civicrm/civicrm-buildkit/blob/11856f8f72f5b24919b358c086cea5f72ab4ed2d/src/jobs/CiviCRM-Distmaker.job#L116-L124)... )

File: distmaker/dists/joomla5_php.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# copy all the rest of the stuff
+dm_reset_dirs "$TRG" "$DM_TMPDIR/com_civicrm"
+cp $SRC/civicrm.config.php $TRG
+dm_generate_version "$TRG/civicrm-version.php" Joomla
+dm_install_core "$SRC" "$TRG"
+dm_install_coreext "$SRC" "$TRG" $(dm_core_exts)
+dm_install_packages "$SRC/packages" "$TRG/packages"
+dm_install_vendor "$SRC/vendor" "$TRG/vendor"
+rm -rf $TRG/vendor/pear/log
```
Comment (seamuslee001) [inline distmaker/dists/joomla5_php.sh:25]: @totten this also needs to change to psr/log not pear/log just FYI

**Comment by aydun** [comment]
joomla5 build has expected files - ie same as 'alt' without `psr/log`

**Pull request merged**